### PR TITLE
Remove code coverage setup

### PR DESCRIPTION
**Pull request by youknowriad**
The Code Coverage was helpful at the early stages of the project where we didn't have tests. It's not anymore as it only counts JavaScript unit tests and we have more tests than that. The results in PR are also not reliable. So let's just remove it to have a lighter setup.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -152,7 +151,7 @@
 		"check-licenses": "concurrently \"wp-scripts check-licenses --prod --gpl2\" \"wp-scripts check-licenses --dev\"",
 		"precheck-local-changes": "npm run docs:build",
 		"check-local-changes": "( git diff -U0 | xargs -0 node bin/process-git-diff ) || ( echo \"There are local uncommitted changes after one or both of 'npm install' or 'npm run docs:build'!\" && exit 1 );",
-		"ci": "concurrently \"npm run lint\" \"npm run test-unit:coverage-ci\" \"npm run check-local-changes\"",
+		"ci": "concurrently \"npm run lint\" \"npm run check-local-changes\"",
```
Comment (gziolo) [inline package.json]: `test-unit:coverage-ci` should still be here we just need to update it as follows:
```json
"test-unit:coverage-ci": "npm run test-unit -- --ci --runInBand",
```

**Review by gziolo — DISMISSED** [review]
Can we leave the setup for test coverage but only don't use it with Travis? In other words, everything that is related to `codecov` tool. I still see value in being able to run `npm run test-unit:coverage` locally.

**Comment by youknowriad** [comment]
@gziolo My opinion is that it useless and just complexify the setup of the repository for nothing. Its results are not reliable and keeping it in the repository while not running in CI will more likely end up unmaintained.

**Comment by gziolo** [comment]
I don't plan for keeping this functionality. If more people find is as useless we can remove it. 

However `run ci` needs to run unit tests :) See: https://github.com/WordPress/gutenberg/pull/11198#discussion_r228889531

**Comment by aduth** [comment]
I'd rather we not commit to maintaining it. It's likely to become neglected anyways if not part of the main pipeline.

File: packages/jest-preset-default/README.md [hunk 2]
```diff
@@ -24,8 +24,6 @@ npm install @wordpress/jest-preset-default --save-dev
 
 #### Brief explanations of options included
 
-* `coverageDirectory` - the directory where Jest outputs its coverage files is set to `coverage`.
```
Comment (gziolo) [inline packages/jest-preset-default/README.md]: We need to update changelog as this looks as a breaking change, or we can keep it as is 😄

**Comment by gziolo** [comment]
LGTM, let's proceed when Travis is green.

**Comment by youknowriad** [comment]
Thanks for your help @desrosj we should update the package CHANGELOG, I'm going to do it right now.

**Comment by youknowriad** [comment]
This should be ready to go I need a ✅

**Review by gziolo — APPROVED** [review]
Sad to see this removed but happy at the same time to see our setup gets simplified

**Pull request merged**